### PR TITLE
syslog-debun: several enhancements

### PR DESCRIPTION
**Pull request by pasztor**
* enhanced process information gathering
* syscall tracing (strace/truss)
* enhanced configuration gathering (--preprocess-into)
* selinux related information gathering
* debian/redhat recognition when lsb_release is not available
* solaris information gathering enhancements (sysdef, kstat, showrev, release)
* get information about syslog-ng svr4 solaris packages, if possible

Signed-off-by: PÁSZTOR György <gyorgy.pasztor@balabit.com>

### CODE REVIEW

**Comment by lbudai** [comment]
hi,

could you provide some unit tests? (shunit2, etc...?)



(I'm just joking :-) ) 

Thanks for the patch! (maybe it's containing too long  functions for my taste, and maybe it could be split into smaller functions but well, if you got another :+1: , it's OK for me... temperature just so high on these days that I won't make a point of doing clean code in a bash script ... )

**Comment by bazsi** [comment]
As this is your original work I would let you determine the coding style.

:+1:

File: contrib/syslog-debun [hunk 1]
```diff
@@ -178,6 +189,15 @@ debun_finish_debug () {
...
 		kill -INT $pcappid
 	fi
+	if [ -n "$tracing" -a -z "$debug_params" ]; then
+		for i in ${tracepids[*]} ; do
+			kill -INT $i
+		done
+		sleep 2
+		for i in ${tracepids[*]} ; do
+			[ -d /proc/${i} ] && kill -9 $i 2>/dev/null
+		done
```
Comment (deirf) [inline contrib/syslog-debun]: It is good.
Question:
Why do not use loop that try kill the process in repeated cycles if it does not die. This technique is implemented in for example some debian stop scripts. After this it is possible that do not have to implement the same function every time if a process is have to killed.

Comment (pasztor) [inline contrib/syslog-debun]: As you can see, in the previous parts of this debun_finish_debug (), I try a more sophisticated way for other process killings.
The strace/truss is a new function: it's a first approach for handling the finishing.
As you can see already appeared some process related function, which is more portable.
But it's a much longer process to write a solution which works everywhere.

Now the primary goal was that, to incorporate the new enhancements as soon as possible.
Refactoring, and maybe unit tests, etc. will happen at a later time.

**Pull request merged**